Design + implement frontend form component

fanzzz368 opened the pull request:
### ℹ️ Issue

Closes #<issue number>

### 📝 Description

Briefly list the changes made to the code:

- Added a basic medical form (the ones signed by medical professionals)
- Added a basic grant form (the one signed by child's guardians/parents)
- Added a medical form page and a grant form page

### ✔️ Verification

What steps did you take to verify your changes work? These should be clear enough for someone to be able to clone the branch and follow the steps themselves. 

- I put the form pages in app.tsx and displayed it on my local host to see what they look like and there were no errors.

Provide screenshots of any new components, styling changes, or pages. 
<img width="1440" alt="Screenshot 2023-03-01 at 6 13 12 PM" src="https://user-images.githubusercontent.com/91427854/222291865-ac4e2878-cf1e-4ea1-8f73-04578efabf1f.png">

image of the grant form


<img width="1423" alt="Screenshot 2023-03-01 at 6 13 49 PM" src="https://user-images.githubusercontent.com/91427854/222291880-204ba1d0-ddb5-4407-886c-71c0c0c0ff61.png">

image of the medical form 

### 🏕️ (Optional) Future Work / Notes

Did you notice anything ugly during the course of this ticket? Any bugs, design challenges, or unexpected behavior? Write it down so we can clean it up in a future ticket!

Future steps:
- put the inputs into a card container and make the formatting better
- add the paragraph of disclaimer at the bottom of the page
- add a footer that contains info abt TSWGO's address, website, and email
- find a way for users to upload a bill for the "intended use of grant" part in MedicalForm

fanzzz368 committed:
Implemented the basics for the form components

fanzzz368 committed:
Revert "Implemented the basics for the form components"

This reverts commit 569d6bd1bbda435fa00b232e7c10898fa317be74.

fanzzz368 committed:
Revert "Revert "Implemented the basics for the form components""

This reverts commit 7bd312e2c4b445a008c7a94477a64e7c3effbb01.

jackielincroft commented on apps/frontend/src/components/form/GrantForm.tsx:
```diff
@@ -0,0 +1,117 @@
+interface GrantFormValues {
+  childName: string;
+  ssn: string;
+  dob: string;
```
can you make this a Date type instead of string?

jackielincroft commented on apps/frontend/src/components/form/GrantForm.tsx:
```diff
@@ -0,0 +1,117 @@
+  grantAmount: string;
+  useOfGrant: string;
+  parentSignature: string;
+  parentDate: string;
```
this might be better named signatureDate (i feel like parentDate is kind of ambiguous/confusing)

jackielincroft replied:
and also Date type instead of string here!

jackielincroft commented on apps/frontend/src/components/header/Header.tsx:
```diff
@@ -0,0 +1,14 @@
+import { Box, Image } from '@chakra-ui/react';
+import Logo from '../header/Swgologo.jpeg';
+
+const Header: React.FC = () => {
```
thanks for adding this header component!

jackielincroft commented on apps/frontend/src/components/form/MedicalForm.tsx:
```diff
@@ -0,0 +1,109 @@
+
+interface MedicalFormValues {
+  childDiagnosis: string;
+  diagnosisDate: string;
```
again, could be type Date

jackielincroft commented on apps/frontend/src/components/form/MedicalForm.tsx:
```diff
@@ -0,0 +1,109 @@
+  doctorsExplanation: string;
+  doctorPrintedNameTitle: string;
+  doctorSignature: string;
+  doctorDate: string;
```
maybe doctorSignatureDate ?

jackielincroft commented on apps/frontend/src/components/form/MedicalForm.tsx:
```diff
@@ -0,0 +1,109 @@
+  childDiagnosis: string;
+  diagnosisDate: string;
+  childPhysician: string;
+  hospital: string;
```
i think hospitalName would be a little clearer

jackielincroft commented on apps/frontend/src/components/form/MedicalForm.tsx:
```diff
@@ -0,0 +1,109 @@
+
+    <InputField
+      isRequired
+      displayName="Parent/Legal Guardian Signature:"
```
should this be doctor signature?  maybe a copy paste error?

jackielincroft commented on apps/frontend/src/components/form/types.ts:
```diff
@@ -0,0 +1,42 @@
+/**
+ * Common props for all form field components.
+ */
+export interface GenericFormFieldProps {
```
do you know where these are used?

fanzzz368 replied:
Ohh, I think GenericInputFieldProps<ValueType> was there before for validating the inputs, but we decided that we didn't need to validate for now. I'll delete this!

fanzzz368 replied:
I used the top interface (GenericFormFieldProps) in InputField.tsx to place some constraints on the user input like I made isRequired true for every input box on the form so the user has to fill out the form completely to submit it. I'm not sure if that's necessary though, let me know if it would be cleaner to delete it

jackielincroft replied:
i think removing it is the right call -- it seems like kind of unnecessary bulk for the simplicity of the form we're working on

jackielincroft commented on apps/frontend/src/components/form/GrantForm.tsx:
```diff
@@ -0,0 +1,117 @@
+  childPhone: string;
+  childCellphone: string;
+  childEmail: string;
+  grantAmount: string;
```
i think maybe this should be a number, and labelled grantDollarAmount

jackielincroft commented on apps/frontend/src/components/form/GrantForm.tsx:
```diff
@@ -0,0 +1,117 @@
+      displayName="SSN:"
+      fieldName="ssn"
+    />
+    <InputField
```
this should probably be a date picker component

jackielincroft commented on apps/frontend/src/components/form/GrantForm.tsx:
```diff
@@ -0,0 +1,117 @@
+      displayName="Phone:"
+      fieldName="childPhone"
+    />
+    <InputField
```
there's a chakra prop to give a field `type="email"` that could be nice to use here

jackielincroft commented on apps/frontend/src/components/form/GrantForm.tsx:
```diff
@@ -0,0 +1,117 @@
+      displayName="E-mail Address:"
+      fieldName="childEmail"
+    />
+    <InputField
```
this should probably be a number picker

jackielincroft commented on apps/frontend/src/components/form/GrantForm.tsx:
```diff
@@ -0,0 +1,117 @@
+    />
+    <InputField
+      isRequired
+      displayName="Requested grant amount:"
```
for clarity, maybe say "Requested grant amount (in USD):"

jackielincroft commented on apps/frontend/src/components/form/GrantForm.tsx:
```diff
@@ -0,0 +1,117 @@
+      displayName="Parent/Legal Guardian Signature:"
+      fieldName="parentSignature"
+    />
+    <InputField
```
again, probably date picker

jackielincroft commented on apps/frontend/src/pages/medicalFormPage.tsx:
```diff
@@ -0,0 +1,23 @@
+        </Heading>
+        <Text textAlign={"center"}>
+        (to be completed by medical professional)
+        </Text>
```
i think we could use a little more padding around the heading and text

jackielincroft reviewed (commented):
thanks for the hard work fanny!  i left a bunch of comments, i think in general having string data types is fine but dates could have very different formats depending what people type so it would be good to standardize those.  i think importing some different chakra components (like this date picker: https://chakra-ui.com/docs/components/input/usage#input-methods-other-than-text and maybe this dollar number picker: https://chakra-ui.com/docs/components/number-input/usage#formatting-and-parsing-the-value) would be helpful here!

also, it looks like from your screenshot the chakra styling isn't actually being applied.  the entire component needs to be wrapped in a <ChakraProvider> component for styling to apply (our entire app will be wrapped in this component!), so if you add that things will look much better :) although we still have some minor spacing to fix, but we can work on that as we go

((for reference, this is what it looks like when chakra styling is being applied!))
<img width="961" alt="Screenshot 2023-03-01 at 9 16 33 PM" src="https://user-images.githubusercontent.com/55663537/222315101-ebc1e8d3-654d-41a3-93a7-5bf26d968947.png">

fanzzz368 committed:
renamed form variables + fixed data types

jackielincroft committed:
clean up unnecessary types/props from input field

jackielincroft committed:
update grant form to use specific inputs for date, number input, etc and put components directly in form

fanzzz368 committed:
Removed Types.tsx + Implement TimeInputField, EmailInputField, and NumberInputField

fanzzz368 committed:
made inputfields fit to screen + added padding around heading and text

fanzzz368 committed:
Merge remote-tracking branch 'origin/jl-form-frontend' into Implemented-Form-Basics

fanzzz368 committed:
Revert "Merge remote-tracking branch 'origin/jl-form-frontend' into Implemented-Form-Basics"

This reverts commit 8ed9cff1d3d8d21c2d38e37c03c445b73ede023e, reversing
changes made to 82dc0b11ab70b5d1c8d829a5f124414f084ecbfb.

fanzzz368 committed:
Revert "Revert "Merge remote-tracking branch 'origin/jl-form-frontend' into Implemented-Form-Basics""

This reverts commit 0c83ab7b2ff91b3d6af1f93902ba213f22f6097b.

fanzzz368 committed:
added padding to header and text + condensed form input fields into one file

fanzzz368 committed:
added phone icon to phone on medical form

fanzzz368 committed:
made only phone numbers, names, and emails required + changed to child diagnosis and hospital to dropdown + removed ssn

fanzzz368 committed:
moved to dropdown values to enum at top of page

Questions:
- there are 2 groups of dropdown values (values for cancers and values for hospitals): do i put them both in the same enum Dropdown or should I put them in two separate groups of enum?
- the input fields have titles above them and the dropdown bars have the titles within the dropdown bar — do we need to be consistent and place title of dropdown bar on top of it too

fanzzz368 committed:
add notes section to the grant form

fanzzz368 committed:
added notes sections for both forms + separated cancers and hospitals enum

jackielincroft commented on apps/frontend/src/components/form/GrantForm.tsx:
```diff
@@ -0,0 +1,283 @@
+        </FormControl>
+      )}
+    </Field>
+    <Field name="dob">
```
sorry i know i didn't say this before, but i think having dob required is probably good

jackielincroft commented on apps/frontend/src/components/form/MedicalForm.tsx:
```diff
@@ -0,0 +1,275 @@
+      )}
+    </Field>
+
+    <Select placeholder="Hospital Name" size='md'>
```
i think this should probably be wrapped in a field/form control like all the other fields and be required!  also, if you could add an optional text entry field below this dropdown that says something like "If Other, please specify" that would be great

jackielincroft commented on apps/frontend/src/components/form/MedicalForm.tsx:
```diff
@@ -0,0 +1,275 @@
+const ExampleForm: React.FC<ExampleFormProps> = ({ onSubmit }) => (
+  <Form onSubmit={onSubmit ?? alertOnSubmit} initialValues={{}}>
+
+    <Select placeholder="Select Child's Diagnosis" size='md'>
```
see comments for the select hospital dropdown!

jackielincroft reviewed (dismissed):
thanks so much for the hard work fanny!! if you could just make those last couple changes to the select menus and then once those are done merge this into main!!  i know there's still some spacing/styling work to clean up but that can be a separate PR since it would be great to get these base forms in so we can start hooking up the frontend to the backend

fanzzz368 committed:
text entry fields for other cancers and other hospitals

fanzzz368 committed:
made dob required + wrapped the select dropdowns in form/form control

jackielincroft reviewed (approved):
🚢 !
